tests/scripts/generate_test_code.py

```diff
@@ -552,14 +552,12 @@ def skip_comments(line, stream):
         pos = closing + 2
         # Replace inner comment by spaces. There needs to be at least one space
         # for things like 'int/*ihatespaces*/foo'. Go further and preserve the
-        # width of the comment, this way column positions in error messages
-        # remain correct.
-        # TODO: It would be better to preserve line breaks, to get accurate
-        # line numbers if there's something interesting after a comment on
-        # the same line.
+        # width of the comment and line breaks, this way positions in error
+        # messages remain correct.
         line = (line[:opening.start(0)] +
-                ' ' * (pos - opening.start(0)) +
+                re.sub(r'.', r' ', line[opening.start(0):pos]) +
```

Comment:
Do line continuations using `\` in `//` comments count as the same line or different lines? I would have thought that it would be useful to treat them as different lines seeing as compiler errors will match up with source lines.